GooglePayment/src/main/java/com/braintreepayments/api/GooglePaymentClient.java

```diff
@@ -67,61 +61,64 @@
     private static final String CARD_PAYMENT_TYPE = "CARD";
     private static final String PAYPAL_PAYMENT_TYPE = "PAYPAL";
 
+    private BraintreeClient braintreeClient;
+
+    public GooglePaymentClient(BraintreeClient braintreeClient) {
+        this.braintreeClient = braintreeClient;
+    }
+
     /**
      * Before starting the Google Payments flow, use this method to check whether the
-     * Google Payment API is supported and set up on the device. When the listener is called with
+     * Google Payment API is supported and set up on the device. When the callback is called with
      * {@code true}, show the Google Payments button. When it is called with {@code false}, display other
      * checkout options.
      *
-     * @param fragment {@link BraintreeFragment}
-     * @param listener Instance of {@link BraintreeResponseListener<Boolean>} to receive the
-     *                 isReadyToPay response.
+     * @param activity {@link FragmentActivity}
+     * @param callback Instance of {@link GooglePaymentIsReadyToPayCallback} to receive the
+     *                 isReadyToPay result.
      */
-    public static void isReadyToPay(final BraintreeFragment fragment,
-                                    final BraintreeResponseListener<Boolean> listener) {
-        isReadyToPay(fragment, null, listener);
+    public void isReadyToPay(final FragmentActivity activity, final GooglePaymentIsReadyToPayCallback callback) {
```

Comment:
Could we be a little more generic here and in the other methods and allow merchants to pass a `Context` instead of a `FragmentActivity`? Or does it specifically need to be a `FragmentActivity`?

Reply:
Agreed yeah. We want to be less restrictive in areas where it can be.

Reply:
There may be an overload for the `Wallet` accessor though that accepts `Context` that we can use. We would have to make a slight change to the merchant facing design, but it looks possible.